### ISSUE TITLE
Hardcoded directories to images, scripts, and styles.

### ISSUE THREAD

**Issue by robhudson** [sunlightlabs/django-mediasync#4]
I consider it a bug that I have to renamed my subdirectories to match what mediasync expects.  I've heard this as a complaint from at least one other person who I recommended this project to.

I did find the undocumented `MEDIA_CSS_PATH` and `MEDIA_JS_PATH`.  Perhaps that should be documented?  I haven't tested it yet.

But I'm curious if those paths are absolutely necessary or if mediasync can deduce them or just not need them specifically?  For example, the sync method can traverse all subdirectories of the media folder; the media tags could possibly have a mapping of base name to full path name that gets used.

**Comment by jcarbaugh** [comment]
I'm about to check in some changes that resolves a portion of the issue. The code that does the syncing now has no knowledge of the expected directory structure. Also, it used to assume that everything was in a media directory under PROJECT_ROOT. It will now just use the MEDIA_ROOT setting.

I like the shortcuts that having MEDIASYNC_CSS_PATH and MEDIASYNC_JS_PATH (I renamed them) allow you. Not having to specify scripts in {% js "scripts/jquery.js" %} is quite useful. However I can see how it could be a problem if you have scripts in multiple folders in your media directory. I think we can accommodate this by just documenting the JS/CSS path settings.

By setting them to empty strings:

```
MEDIASYNC_CSS_PATH = ""
MEDIASYNC_JS_PATH = ""
```

This essentially sets both paths to MEDIA_ROOT. You can then specify full paths in the template tags.

```
{% js "js-external/jquery.js" %}
{% js "js-internal/site.js" %}
```

Would documenting this solve the problem?

**Comment by robhudson** [comment]
Yes, that would make things a lot better.

I start to wonder, however, if the code would be simpler and clearer if there were no special PATHs at all.  Aside from the shortcut, do they do anything else?

I'll be using this regardless, but how about some devil's advocate stuff just in case.  :)

By defaulting to non-empty values, any project that has different paths has extra steps to get mediasync working, and extra things to understand about it.  I like opting in to shortcuts rather than opting out of them.

`{% js "path_to_scripts/jquery.js" %}` is a natural extension of the what you'd do in normal Django templates, as well... `<script src="{{ MEDIA_URL }}/path_to_scripts/jquery.js"></script>`.  It's less magical and more explicit.

**Comment by jcarbaugh** [comment]
Your logic has persuaded me. I'll make MEDIASYNC_CSS_PATH and MEDIASYNC_JS_PATH default to empty paths and allow the user to override if they want to use the shortcuts. Those defaults should be the least surprising to people.

**Comment by jcarbaugh** [comment]
Default to empty CSS and JS paths. Document the usage of MEDIASYNC_CSS_PATH and MEDIASYNC_JS_PATH. Closed by 662a3c41b5297c96498ab6de0fdd820a9112b06f.
